contrib/platform/src/com/sun/jna/platform/unix/LibCAPI.java

```diff
@@ -224,17 +232,144 @@ public ssize_t(long value) {
+     * If execution of the new file succeeds, it updates the access time field of the file as if the file had been read.
+     * @see <a href="https://www.freebsd.org/cgi/man.cgi?query=execve">execve(3)</a>
+     */
+    int execve(String filename, String[] argv, String[] env);
+
+    /**
+     * Similar to {@link #execv(String, String[])}, but instead of identifying the program executable by its pathname,
+     * the file descriptor fd is used.
+     * <p>
+     * On Linux, fexecve can fail with an error of ENOSYS if /proc has not been mounted and
+     * the kernel lacks support for the underlying execveat system call.
+     *
+     * @param fd   file descriptor. Must have been opened with the O_RDONLY flag or (on Linux) the O_PATH flag.
+     * @param argv array that is used to provide a value for the argv argument to the main function of the program to be executed.
+     *             The last element of this array must be a null pointer. By convention, the first element of this array is the file name of the program sans directory names.
+     * @param env  the enviornment variables.
+     * @return normally doesn't return, since execution of a new program causes the currently executing program to go away completely.
+     * A value of -1 is returned in the event of a failure or 145 (E2BIG), 130 (ENOEXEC) or 132 (ENOMEM), in addition to the usual file name errors.
+     * If execution of the new file succeeds, it updates the access time field of the file as if the file had been read.
+     * @see <a href="https://www.freebsd.org/cgi/man.cgi?query=fexecve">fexecve(3)</a>
+     */
+    int fexecve(int fd, String[] argv, String[] env);
+
+    /**
+     * The waitpid function is used to request status information from a child process.
+     * Normally, the calling process is suspended until the child process makes status information available by terminating.
+     * <p>
+     * If status information for a child process is available immediately, this function returns immediately without waiting.
+     * If more than one eligible child process has status information available, one of them is chosen randomly, and its status is returned immediately.
+     * To get the status from the other eligible child processes, you need to call waitpid again.
+     * <p>
+     * This function is a cancellation point in multi-threaded programs.
+     * This is a problem if the thread allocates some resources (like memory, file descriptors, semaphores or whatever) at the time waitpid is called.
+     * If the thread gets canceled these resources stay allocated until the program ends.
+     * To avoid this calls to waitpid should be protected using cancellation handlers.
+     * @param pid       the process ID of the child process to request information from.
+     *                  Can also be -1 or WAIT_ANY, which requests status information for any child process.
```

Comment:
I haven't looked, but try to avoid assigning numbers, assuming `WAIT_ANY` is -1.  If the constant is cross-platform, define it as such and `@link` to the constant.  Otherwise just use `@code` tags for the error name without defining its number.